Bugfix/1305 expansion test flakiness

jbertran opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

Flaky test management time!

* Some pods fail to start even after 90s (30 x 3s wait).
Example build: https://eve.devsca.com/github/scality/metalk8s/#/builders/1/builds/6895

* More stuff...

**Summary**:

* Revert expansion test disable
* Change approach for command execution in salt-master
* Enable log collection on `node1`, since we now do expansion

**Acceptance criteria**: 

* No more flakies due warning logs going into `stderr` in salt output of `deploy_node`.
* Actual formula execution errors still report failures (see https://eve.devsca.com/github/scality/metalk8s/#/builders/10/builds/2300 for example, where a syntax error inserted in the `deploy_node` Jinja file caused the build to fail with comprehensive error output).


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #1305 
<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

jbertran committed:
Revert "Disable flaky expansion test"

This reverts commit a9e3676d685d4bca1218394e0ac0f673d8e660ce.

Issue: #1305

bert-e commented:
# Hello jbertran,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/1305-expansion-test-flakiness` into
`development/2.0`
* [w/2.1/bugfix/1305-expansion-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/1305-expansion-test-flakiness) will be merged into `development/2.1` 
* [w/2.2/bugfix/1305-expansion-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/1305-expansion-test-flakiness) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @nootal

* @cmonfort

* @lucieleonard

* @alxf

* @carlito-scality

* @alexandre-allard-scality

* @teddyandrieux

* @slaperche-scality

* @thomasdanan

* @zempashi

* @nicolast

* @sayf-eddine-scality

* @monpote

* @gdemonet

* @tcarmet

bert-e commented:
# History mismatch



Merge commit #a11b2d3df10f4f1ec6dbf8d83c6ec7efab078dcb on the integration branch
`w/2.1/bugfix/1305-expansion-test-flakiness` is merging a branch which is neither the current
branch `bugfix/1305-expansion-test-flakiness` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/1305-expansion-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

jbertran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/1305-expansion-test-flakiness` into
`development/2.0`
* [w/2.1/bugfix/1305-expansion-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/1305-expansion-test-flakiness) will be merged into `development/2.1` 
* [w/2.2/bugfix/1305-expansion-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/1305-expansion-test-flakiness) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @nootal

* @cmonfort

* @lucieleonard

* @alxf

* @carlito-scality

* @alexandre-allard-scality

* @teddyandrieux

* @slaperche-scality

* @thomasdanan

* @zempashi

* @nicolast

* @sayf-eddine-scality

* @monpote

* @gdemonet

* @tcarmet

jbertran committed:
tests: switch to `kubectl` shell calls for salt commands

This allows us to examine the return code of the salt commands, provided
by kubectl but oddly not by the API.

Closes: #1305 #1331

bert-e commented:
# History mismatch



Merge commit #0a7e87c66680ec7065d22e704ee171fefd11af73 on the integration branch
`w/2.1/bugfix/1305-expansion-test-flakiness` is merging a branch which is neither the current
branch `bugfix/1305-expansion-test-flakiness` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/1305-expansion-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

jbertran committed:
ci: collect logs on node1 after expansion tests

Since we activate expansion tests, we can now target node1 for sosreport
log collection and upload.

Issue: #1305

slaperche-scality commented on tests/install/steps/test_expansion.py:
```diff
@@ -115,7 +115,7 @@ def kubectl_exec(
         kube_args.extend(['-n', kwargs.get('namespace')])
 
-    kubectl_cmd_tplt = 'kubectl exec %s %s -- %s'
+    kubectl_cmd_tplt = 'kubectl exec {} {} -- {}'
```
This doesn't belong to this commit.

jbertran committed:
Revert "Disable flaky test for static pods management"

This reverts commit d26e35655736c88ee58c9f63c2c9aa65564f416a.

Closes: #1320

jbertran commented:
@bert-e approve

bert-e commented:
# History mismatch



Merge commit #0a7e87c66680ec7065d22e704ee171fefd11af73 on the integration branch
`w/2.1/bugfix/1305-expansion-test-flakiness` is merging a branch which is neither the current
branch `bugfix/1305-expansion-test-flakiness` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/1305-expansion-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

jbertran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/1305-expansion-test-flakiness` into
`development/2.0`
* [w/2.1/bugfix/1305-expansion-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/1305-expansion-test-flakiness) will be merged into `development/2.1` 
* [w/2.2/bugfix/1305-expansion-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/1305-expansion-test-flakiness) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/9d29135ddf03b5326244d65f28d85f52349ff2c9) did not succeed in branch bugfix/1305-expansion-test-flakiness.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/7d2d342bc0f7c1814df38706f544c0790158ec88) did not succeed in branch w/2.2/bugfix/1305-expansion-test-flakiness.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye jbertran.
